test/utils/test_device.py

```diff
@@ -1,28 +1,120 @@
+import os
 from unittest.mock import patch
 
-from haystack.utils import get_device
+import pytest
 
+from haystack.utils import ComponentDevice, Device, DeviceMap, DeviceType
 
-@patch("torch.cuda.is_available")
-def test_get_device_cuda(torch_cuda_is_available):
-    torch_cuda_is_available.return_value = True
-    device = get_device()
-    assert device == "cuda:0"
+
+def test_device_type():
+    for e in DeviceType:
+        assert e == DeviceType.from_str(e.value)
+
+    with pytest.raises(ValueError, match="Unknown device type string"):
+        DeviceType.from_str("tpu")
```

Comment:
Let's parametrize this test instead of using a for loop

Reply:
Not sure I follow? The current code will always check for all enum variants exhaustively. Why would we want to parameterize it?

Reply:
In general I prefer to parametrize tests instead of using loops because, when pytest is configured to run tests in parallel, the parametrization can make the execution quite a bit faster. In here it's completely negligible though, so no problem, let's leave it like this.